WalletWasabi.Tests/UnitTests/Crypto/ProofSystemTests.cs

```diff
@@ -41,6 +44,76 @@ public void CanProveAndVerifyMAC()
 			var clientStatement = ProofSystem.IssuerParameters(coordinatorParameters, mac, Ma);
 			var isValidProof = ProofSystem.Verify(clientStatement, proofOfMac);
 
+			Assert.True(isValidProof);
+			////////
+
+			var corruptedResponses = new ScalarVector(proofOfMac.Responses.Reverse());
+			var invalidProofOfMac = new Proof(proofOfMac.PublicNonces, corruptedResponses);
+			isValidProof = ProofSystem.Verify(clientStatement, invalidProofOfMac);
+
+			Assert.False(isValidProof);
+
+			var corruptedPublicNonces = new GroupElementVector(proofOfMac.PublicNonces.Reverse());
+			invalidProofOfMac = new Proof(corruptedPublicNonces, proofOfMac.Responses);
+			isValidProof = ProofSystem.Verify(clientStatement, invalidProofOfMac);
+
+			Assert.False(isValidProof);
+		}
+
+		[Fact]
+		[Trait("UnitTest", "UnitTest")]
+		public void CanProveAndVerifySerialNumbers()
+		{
+			var rnd = new SecureRandom();
+			var z = rnd.GetScalar();
+			var r = rnd.GetScalar();
+			var a = rnd.GetScalar();
+
+			var witness = new ScalarVector(z, a, r);
+			var Ca = z * Generators.Ga + r * Generators.Gh + a * Generators.Gg;
+			var S = r * Generators.Gs;
+
+			var statement = ProofSystem.SerialNumber(Ca, S);
+			var proofOfSerialNumber = ProofSystem.Prove(new Knowledge(statement, witness), rnd);
+
+			// The coordinator must verify the blinded amout is zero
```

Comment:
i don't think this comment is accurate, or maybe i'm misunderstanding it

Reply:
you're right. It was because of copy pasting